lib/improver/weighted_blend.py

```diff
@@ -0,0 +1,111 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing Weighted Blend classes."""
+import warnings
+
+import numpy as np
+import iris
+
+
+class BasicWeightedAverage(object):
+    """Apply a Basic Weighted Average to a cube."""
+
+    def __init__(self, coord, coord_adjust=None):
+        """Set up for a Basic Weighted Average Blending plugin
+
+        Args:
+            coord : string
+                     The name/s of a coordinate dimension/s in the cube
+            coord_adjust : Function to apply to the coordinate after
+                           collapsing the cube to correct the values
+                           for example for time windowing and
+                           cycle averaging the follow function would
+                           adjust the time coordinates
+            e.g. coord_adjust = lambda pnts: pnts[len(pnts)/2]
+        """
+        self.coord = coord
+        self.coord_adjust = coord_adjust
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return (
+            '<BasicWeightedAverage: coord = {0:s}>').format(self.coord)
+
+    def process(self, cube, weights=None):
+        """Calculated weighted mean across the chosen coord
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+
+        Returns:
+            result : iris.cube.Cube
+
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            raise ValueError('The first argument must be an instance of ' +
```

Comment:
Typically, we've been defining a `msg` variable with the message, and then raising the error with `raise ValueError(msg)`, although this may be a stylistic preference.